locations/jclouds/src/test/java/org/apache/brooklyn/location/jclouds/JcloudsByonLocationResolverStubbedTest.java

```diff
@@ -0,0 +1,157 @@
+
+    @SuppressWarnings("unused")
+    private static final Logger log = LoggerFactory.getLogger(JcloudsByonLocationResolverStubbedTest.class);
+    
+    private final String nodeId = "mynodeid";
+    private final String nodePublicAddress = "173.194.32.123";
+    private final String nodePrivateAddress = "172.168.10.11";
+    
+    protected LocalManagementContext newManagementContext() {
+        // This really is stubbed; no live access to the cloud
+        LocalManagementContext result = LocalManagementContextForTests.builder(true).build();
+        BrooklynProperties brooklynProperties = result.getBrooklynProperties();
+        brooklynProperties.put("brooklyn.location.jclouds."+SOFTLAYER_PROVIDER+".identity", "myidentity");
+        brooklynProperties.put("brooklyn.location.jclouds."+SOFTLAYER_PROVIDER+".credential", "mycredential");
+        return result;
+
+    }
+
+    @Override
+    protected NodeCreator newNodeCreator() {
+        return new NodeCreator() {
+            @Override
+            public Set<? extends NodeMetadata> listNodesDetailsMatching(Predicate<ComputeMetadata> filter) {
+                NodeMetadata result = new NodeMetadataBuilder()
+                        .id(nodeId)
+                        .credentials(LoginCredentials.builder().identity("dummy").credential("dummy").build())
+                        .loginPort(22)
+                        .status(Status.RUNNING)
+                        .publicAddresses(ImmutableList.of(nodePublicAddress))
+                        .privateAddresses(ImmutableList.of(nodePrivateAddress))
+                        .build();
+                return ImmutableSet.copyOf(Iterables.filter(ImmutableList.of(result), filter));
+            }
+            @Override
+            protected NodeMetadata newNode(String group, Template template) {
+                throw new UnsupportedOperationException();
+            }
+        };
+    }
+
+    @Test
+    public void testResolvesHostInSpec() throws Exception {
+        String spec = "jcloudsByon:(provider=\""+SOFTLAYER_PROVIDER+"\",region=\""+SOFTLAYER_AMS01_REGION_NAME+"\",user=\"myuser\",password=\"mypassword\",hosts=\""+nodeId+"\")";
+        Map<?,?> specFlags = ImmutableMap.of(JcloudsLocationConfig.COMPUTE_SERVICE_REGISTRY, computeServiceRegistry);
+
+        FixedListMachineProvisioningLocation<MachineLocation> location = getLocationManaged(spec, specFlags);
+        
+        JcloudsSshMachineLocation machine = (JcloudsSshMachineLocation) Iterables.getOnlyElement(location.getAllMachines());
+        assertEquals(machine.getJcloudsId(), nodeId);
+        assertEquals(machine.getPublicAddresses(), ImmutableSet.of(nodePublicAddress));
+        assertEquals(machine.getPrivateAddresses(), ImmutableSet.of(nodePrivateAddress));
+        
+        // TODO what user/password should we expect? Fails below because has "dummy":
+//        assertEquals(machine.getUser(), "myuser");
```

Comment:
This is whats throwing the NPE mentioned earlier, previously it was these user/password values that were being used.